Editor/Scripts/CodeGenerators/GenerateRGActionClasses.cs

```diff
@@ -0,0 +1,257 @@
+                            )
+                        )
+                    );
+
+                // Create the delegate creation expression
+                var delegateCreationExpression = SyntaxFactory.ObjectCreationExpression(actionType)
+                    .WithArgumentList(
+                        SyntaxFactory.ArgumentList(
+                            SyntaxFactory.SingletonSeparatedList(
+                                SyntaxFactory.Argument(methodExpression)
+                            )
+                        )
+                    );
+
+                return SyntaxFactory.Argument(delegateCreationExpression);
+            }
+        }
+
+        private static MemberDeclarationSyntax GenerateStartActionMethod(RGActionInfo action)
+        {
+            var methodParameters = new List<ParameterSyntax>
+            {
+                SyntaxFactory.Parameter(SyntaxFactory.Identifier("input"))
+                             .WithType(SyntaxFactory.ParseTypeName("Dictionary<string, object>"))
+            };
+
+            var parameterParsingStatements = new List<StatementSyntax>();
+            var methodInvocationArguments = new List<string>();
+
+            foreach (var parameter in action.Parameters)
+            {
+                string paramName = parameter.Name;
+                string paramType = parameter.Type;
+
+                methodInvocationArguments.Add(paramName);
+
+                parameterParsingStatements.Add(SyntaxFactory.ParseStatement($"string {paramName}Input = input[\"{paramName}\"] as string;"));
+                parameterParsingStatements.Add(SyntaxFactory.ParseStatement($"{paramType} {paramName} = default;"));
+
+                string tryParseStatement;
+
+                if (paramType.ToLower() == "string" || paramType.ToLower() == "system.string")
+                {
+                    tryParseStatement = $"{paramName} = {paramName}Input;";
+                }
+                else if (RGUtils.IsCSharpPrimitive(paramType))
+                {
+                    tryParseStatement = $"{paramType}.TryParse({paramName}Input, out {paramName});";
+                }
+                else
+                {
+                    tryParseStatement = $"{paramName} = RGSerialization.Deserialize_{paramType.Replace(".", "_")}({paramName}Input);";
+                }
+
+                var tryBlock = SyntaxFactory.Block(SyntaxFactory.SingletonList<StatementSyntax>(
+                    SyntaxFactory.ParseStatement(tryParseStatement)
+                ));
+
+                var catchBlock = SyntaxFactory.CatchClause()
+                                              .WithDeclaration(SyntaxFactory.CatchDeclaration(SyntaxFactory.ParseTypeName("Exception")))
+                                              .WithBlock(SyntaxFactory.Block(SyntaxFactory.SingletonList<StatementSyntax>(
+                                                  SyntaxFactory.ParseStatement($"RGDebug.LogError(\"Failed to parse '{paramName}'\");")
```

Comment:
Can we also add the exception message / stack trace to the log?

Reply:
Fixed in latest commit